docs/_posts/examples/3400-01-31-animate-a-line.html

```diff
@@ -0,0 +1,128 @@
+--- layout: example category: example title: Animate a line description: Animate a line by updating a GeoJSON source on each frame. tags: - layers - sources ---
+<style>
+button {
+  position: absolute;
+  margin: 20px;
+}
+
+#play::after {
+  content: 'Stop';
+}
+
+#play.pause::after {
+  content: 'Play';
+}
+</style>
+
+<div id='map'></div>
+<button id='play' class='pause'></button>
+
+<script>
+mapboxgl.accessToken = 'pk.eyJ1IjoieXVuamllbGkiLCJhIjoiY2lxdmV5MG5rMDAxNmZta3FlNGhyMmpicSJ9.CTEQgAyZGROcpJouZuzJyA';
+var map = new mapboxgl.Map({
+  container: 'map',
+  style: 'mapbox://styles/mapbox/streets-v8',
+  center: [0, 0],
+  zoom: 1
+});
+
+// Create a GeoJSON source with an empty lineString.
+var geojson = {
+  "type": "FeatureCollection",
+  "features": [{
+    "type": "Feature",
+    "geometry": {
+      "type": "LineString",
+      "coordinates": [
+        [0, 0]
+      ]
+    }
+  }]
+};
+
+map.on('load', function() {
+
+  // Add a source and layer displaying the line which will be
+  // animated in a circle as a sine wave along the map.
+  map.addSource('line-animation', {
+    'type': 'geojson',
+    'data': geojson
+  });
+
+  map.addLayer({
+    'id': 'line-animation',
+    'type': 'line',
+    'source': 'line-animation',
+    'layout': {
+      'line-cap': 'round',
+      'line-join': 'round'
+    },
+    'paint': {
+      'line-color': '#3cc',
+      'line-width': 5,
+      'line-opacity': .4
+    }
+  });
+
+  var animation;
+  // timestamp of the moment the animation begins
+  // in a format like 145056.405
+  var startTime;
+  // progress = time - startTime
+  var progress = 0;
+  // number of frames per longitude degree
+  var speedFactor = 40;
+
+  function animateLine(time) {
+
+    // if animation completes a lap of 360 degrees in longitude, reset the animation
+    if (progress > speedFactor * 360) {
+      startTime = time;
+      // reset the geojson to the initial empty state
+      geojson = {
+        "type": "FeatureCollection",
+        "features": [{
+          "type": "Feature",
+          "geometry": {
+            "type": "LineString",
+            "coordinates": [
+              [0, 0]
+            ]
+          }
+        }]
+      };
+    }
+
+    progress = time - startTime;
```

Comment:
This is the source of the inactive tab bug-- when on another tab, requestAnimationFrame does not get called, so your progress variable makes a big jump when you return and you don't add any points in between. Consider something like this to limit the maximum timestep between frames:

    var startTime;
    var lastTime;
    var progress = 0;

    function animateLine(time) {
      var dT = time - lastTime;
      if(dT > 30) dT = 30; // Don't jump too far between animation frames, even if the browser tab is inactive
      progress += dT;
      lastTime = time;
   
      // The rest of what you have should work the same

    }